system_metrics_collector/CMakeLists.txt

```diff
@@ -35,6 +35,7 @@ find_package(rcutils REQUIRED)
 
 add_library(system_metrics_collector SHARED
   src/moving_average_statistics/moving_average.cpp
+  src/moving_average_statistics/types.cpp
   src/moving_average_statistics/types.hpp
```

Comment:
The `.hpp` file should not be here.

Reply:
https://stackoverflow.com/questions/36174499/why-add-header-files-into-add-library-add-executable-command-in-cmake

Some people prefer that style because then you get the headers in the auto-generated IDE solutions.

Reply:
I don't think that's the intention here, otherwise this is a consistency issue, because why would this be the only header file under `add_library()`?